lib/classes/ConfigSchemaHandler/normalizeAjvErrors.test.js

```diff
@@ -0,0 +1,563 @@
+      "functions.someFunc.events[3].http should have required property 'path'",
+
+      // TODO: add list of allowed values as AJV error contains this list in error object
+      'functions.someFunc.events[4].http.method should be equal to one of the allowed values',
+
+      'Event should contain only one root property, but got 3 (http, path, method) at functions.someFunc.events[5].http',
+    ];
+
+    schema = {
+      type: 'object',
+      properties: {
+        functions: {
+          type: 'object',
+          patternProperties: {
+            '^[a-zA-Z0-9-_]+$': {
+              type: 'object',
+              properties: {
+                events: {
+                  type: 'array',
+                  items: {
+                    anyOf: [
+                      {
+                        type: 'object',
+                        properties: {
+                          http: {
+                            anyOf: [
+                              {
+                                type: 'string',
+                                pattern: '^(get|post|put) [a-zA-Z0-9]+$',
+                              },
+                              {
+                                type: 'object',
+                                properties: {
+                                  method: { enum: ['get', 'post', 'put'] },
+                                  path: { type: 'string' },
+                                },
+                                required: ['method', 'path'],
+                                additionalProperties: false,
+                              },
+                            ],
+                          },
+                        },
+                        required: ['http'],
+                        additionalProperties: false,
+                      },
+                    ],
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const validate = ajv.compile(schema);
+    validate(userConfig);
+    ajvErrors = validate.errors;
+    expect(normalizeAjvErrors(ajvErrors, userConfig).map(err => err.message)).to.deep.equal(
+      expectedErrorMessages
+    );
+  });
+
+  it('should show error messages for unsupported function name', () => {
+    userConfig = {
+      functions: {
+        $omeFunc: {
+          handler: 'handler.main',
+        },
+      },
+    };
+    schema = {
+      type: 'object',
+      properties: {
+        functions: {
+          type: 'object',
+          patternProperties: {
+            '^[a-zA-Z0-9-_]+$': {
+              type: 'object',
+            },
+          },
+          additionalProperties: false,
+        },
+      },
+    };
+    const validate = ajv.compile(schema);
+    validate(userConfig);
+    ajvErrors = validate.errors;
+    expect(normalizeAjvErrors(ajvErrors, userConfig).map(err => err.message)).to.deep.equal([
+      "Function name '$omeFunc' must be alphanumeric at functions.",
```

Comment:
What is the original message as returned by AJV? Isn't it good enough?

Reply:
We also should strip trailing `.` from path

Reply:
The original message is `service.name should match pattern "^[a-zA-Z][0-9a-zA-Z-]+$"`

Let me know what you prefer more

Reply:
The original message is `Unrecognized property '$omeFunc' on 'functions'`

Reply:
So I am thinking the original message is good enough. Let me know if you'd want it customized instead

Reply:
I think we definitely should stick to original message.

I think the only case where message requires customization (in sense of being totally replaced) is when we deal with unsupported event (or do you think there are some other cases)?

Reply:
There is a case with improper indentation, we're discussing it in another conversation. 

One downside of this original message is that it does not give user a hint on how to fix it. The custom one tells it is not alphanumeric. But eventually, a user may guess what's wrong

Reply:
> One downside of this original message is that it does not give user a hint on how to fix it. 

Indeed in this case _Unrecognized property_ is confusing, and I think we already discussed in the past that it'll be nicer to have something more clear. 

Nicest would be to force AJV to throw something as _"Property name at functions.$someFunc doesn't match expected format"_ but I think there's no way to achieve that (it's doable for property values but not for names).

Therefore ok, let's keep this customization in, sorry for confusion.

Reply:
I'll remove the dot `.` then